### ISSUE TITLE
Argument 1 passed to Inspector\Models\Segment::__construct() must be an instance of Inspector\Models\Transaction, null given

### ISSUE THREAD

**Issue by PauloArthur** [inspector-apm/inspector-php#6]
**Describe the bug**

The following error is shown in the /storage/logs/laravel.log when opening the nova dashboard on production

In Segment.php line 18:

`Argument 1 passed to Inspector\Models\Segment::__construct() must be an instance of Inspector\Models\Transaction, null given, called in inspector-apm/inspector-php/src/Inspector.php on line 172 {"userId":1,"exception":"[object] (TypeError(code: 0): Argument 1 passed to Inspector\\Models\\Segment::__construct() must be an instance of Inspector\\Models\\Transaction, null given, called in inspector-apm/inspector-php/src/Inspector.php on line 172 at inspector-apm/inspector-php/src/Models/Segment.php:18)`

**Expected behavior**

Work fine as it was before the update.

**Versions**

inspector-apm/inspector-laravel 4.6.21
inspector-apm/inspector-php 3.5.17
laravel/framework ^7.0
laravel/nova "*"

**Works downgrading to:**

inspector-apm/inspector-laravel 4.6.13
inspector-apm/inspector-php 3.5.10

Is this something on be aware on the nova end?

**Comment by ilvalerione** [comment]
Thank you for the report, I'm investigating.

**Comment by ilvalerione** [comment]
Hi @PauloArthur, can you attach the full stactrace?

**Comment by ilvalerione** [comment]
Hi @PauloArthur, the error is caused by an improper use of this package in the inspector-laravel packge that has this repo as dependency.

Can you confirm that you also are using the Laravel package?

**Comment by PauloArthur** [comment]
Hey @ilvalerione, thanks for the reply!

Yes, I am using the inspector-apm/inspector-laravel package.

Btw this is the full stack trace:
```
production.ERROR: Argument 1 passed to Inspector\Models\Segment::__construct() must be an instance of 
Inspector\Models\Transaction, null given, called in /vendor/inspector-apm/inspector-php/src/Inspector.php 
on line 172 {"userId":1,"exception":"[object] (TypeError(code: 0): Argument 1 passed 
to Inspector\\Models\\Segment::__construct() must be an instance of Inspector\\Models\\Transaction, null given, 
called in /vendor/inspector-apm/inspector-php/src/Inspector.php on line 172 at 
/vendor/inspector-apm/inspector-php/src/Models/Segment.php:18)
[stacktrace]
#0 /vendor/inspector-apm/inspector-php/src/Inspector.php(172): Inspector\\Models\\Segment->__construct()
#1 /vendor/inspector-apm/inspector-php/src/Inspector.php(191): Inspector\\Inspector->startSegment()
#2 /vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(261): Inspector\\Inspector->addSegment()
#3 /vendor/inspector-apm/inspector-laravel/src/Views/ViewEngineDecorator.php(48): Illuminate\\Support\\Facades\\Facade::__callStatic()
#4 /vendor/laravel/framework/src/Illuminate/View/View.php(139): Inspector\\Laravel\\Views\\ViewEngineDecorator->get()
#5 /vendor/laravel/framework/src/Illuminate/View/View.php(122): Illuminate\\View\\View->getContents()
#6 /vendor/laravel/framework/src/Illuminate/View/View.php(91): Illuminate\\View\\View->renderContents()
#7 /vendor/laravel/framework/src/Illuminate/Http/Response.php(62): Illuminate\\View\\View->render()
#8 /vendor/laravel/framework/src/Illuminate/Http/Response.php(34): Illuminate\\Http\\Response->setContent()
#9 /vendor/laravel/framework/src/Illuminate/Routing/Router.php(759): Illuminate\\Http\\Response->__construct()
#10 /vendor/laravel/framework/src/Illuminate/Routing/Router.php(731): Illuminate\\Routing\\Router::toResponse()
#11 /vendor/laravel/framework/src/Illuminate/Routing/Router.php(685): Illuminate\\Routing\\Router->prepareResponse()
#12 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}()
#13 /nova/src/Http/Middleware/Authorize.php(18): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#14 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Laravel\\Nova\\Http\\Middleware\\Authorize->handle()
#15 /nova/src/Http/Middleware/BootTools.php(20): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#16 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Laravel\\Nova\\Http\\Middleware\\BootTools->handle()
#17 /nova/src/Http/Middleware/DispatchServingNovaEvent.php(20): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#18 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Laravel\\Nova\\Http\\Middleware\\DispatchServingNovaEvent->handle()
#19 /vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(41): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#20 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Routing\\Middleware\\SubstituteBindings->handle()
#21 /vendor/laravel/framework/src/Illuminate/Auth/Middleware/Authenticate.php(44): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#22 /nova/src/Http/Middleware/Authenticate.php(31): Illuminate\\Auth\\Middleware\\Authenticate->handle()
#23 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Laravel\\Nova\\Http\\Middleware\\Authenticate->handle()
#24 /vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(78): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#25 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle()
#26 /vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#27 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle()
#28 /vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#29 /vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(62): Illuminate\\Session\\Middleware\\StartSession->handleStatefulRequest()
#30 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Session\\Middleware\\StartSession->handle()
#31 /vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#32 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle()
#33 /vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#34 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle()
#35 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#36 /vendor/laravel/framework/src/Illuminate/Routing/Router.php(687): Illuminate\\Pipeline\\Pipeline->then()
#37 /vendor/laravel/framework/src/Illuminate/Routing/Router.php(662): Illuminate\\Routing\\Router->runRouteWithinStack()
#38 /vendor/laravel/framework/src/Illuminate/Routing/Router.php(628): Illuminate\\Routing\\Router->runRoute()
#39 /vendor/laravel/framework/src/Illuminate/Routing/Router.php(617): Illuminate\\Routing\\Router->dispatchToRoute()
#40 /vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(165): Illuminate\\Routing\\Router->dispatch()
#41 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}()
#42 /nova/src/Http/Middleware/ServeNova.php(26): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#43 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Laravel\\Nova\\Http\\Middleware\\ServeNova->handle()
#44 /vendor/inspector-apm/inspector-laravel/src/Middleware/WebRequestMonitoring.php(35): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#45 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Inspector\\Laravel\\Middleware\\WebRequestMonitoring->handle()
#46 /vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#47 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#48 /vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#49 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#50 /vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#51 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle()
#52 /vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php(63): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#53 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\CheckForMaintenanceMode->handle()
#54 /vendor/fruitcake/laravel-cors/src/HandleCors.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#55 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fruitcake\\Cors\\HandleCors->handle()
#56 /vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#57 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fideloper\\Proxy\\TrustProxies->handle()
#58 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#59 /vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(140): Illuminate\\Pipeline\\Pipeline->then()
#60 /vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(109): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
#61 /public/index.php(55): Illuminate\\Foundation\\Http\\Kernel->handle()
#62 {main}
"}
```

**Comment by ilvalerione** [comment]
I released the new version of the inspector-laravel package that fix the issue (4.6.23).

Thank you for the report.
